drivers/dac/ad3530r/ad3530r.h

```diff
@@ -0,0 +1,390 @@
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+/**
+ * @enum ad3530r_id
+ * @brief Device IDs
+ */
+enum ad3530r_id {
+	AD3530R_ID
+};
+
+/**
+ * @enum ad3530r_ch_vref_select
+ * @brief Channel voltage reference options
+ */
+enum ad3530r_ch_vref_select {
+	/* External source with Vref I/O as input */
+	AD3530R_EXTERNAL_VREF_PIN_INPUT,
+	/* Internal source with Vref I/O at 2.5V */
+	AD3530R_INTERNAL_VREF_PIN_2P5V,
+};
+
+enum ad3530r_status {
+	/* Status bits */
+	AD3530R_DEVICE_NOT_READY = 0x0001,
+	AD3530R_INTERFACE_NOT_READY = 0x0002,
+	AD3530R_RESET_STATUS = 0x0004,
+	AD3530R_DAC_UPDATE_STATUS = 0x0008,
+
+	/* Errors */
+	AD3530R_PARTIAL_REGISTER_ACCESS = 0x0002,
+	AD3530R_INVALID_OR_NO_CRC = 0x0008,
+	AD3530R_CLOCK_COUNTING_ERROR = 0x0010,
+	AD3530R_DEVICE_NOT_READY_ERR = 0x0080,
+};
+
+/**
+ * @enum ad3530r_ch_output_range
+ * @brief Channel output range options
+ */
+enum ad3530r_ch_output_range {
+	/* Range from 0 V to VREF */
+	AD3530R_CH_OUTPUT_RANGE_0_VREF,
+	/* Range from 0 V to 2*VREF */
+	AD3530R_CH_OUTPUT_RANGE_0_2VREF,
+};
+
+/**
+ * @enum ad3530r_operating_mode
+ * @brief Operating modes
+ */
+enum ad3530r_operating_mode {
+	AD3530R_CH_OPERATING_MODE_0,
+	AD3530R_CH_OPERATING_MODE_1,
+	AD3530R_CH_OPERATING_MODE_2,
+	AD3530R_CH_OPERATING_MODE_3,
+};
+
+/**
+ * @enum ad3530r_write_mode
+ * @brief Write modes
+ */
+enum ad3530r_write_mode {
+	/* Write to DAC registers. No need to trigger LDAC */
+	AD3530R_WRITE_DAC_REGS,
+	/* Write to input registers. User needs to trigger LDAC */
+	AD3530R_WRITE_INPUT_REGS,
+	/* Write to input registers. LDAC is triggered by the driver */
+	AD3530R_WRITE_INPUT_REGS_AND_TRIGGER_LDAC
+};
+
+/**
+ * @enum ad3530r_mux_out_select
+ * @brief Signals to monitor on MUX_OUT pin
+ */
+enum ad3530r_mux_out_select {
+	POWERED_DOWN,
+	VOUT0,
+	IOUT0_SOURCE_MODE,
+	IOUT0_SINK_MODE,
+	VOUT1,
+	IOUT1_SOURCE_MODE,
+	IOUT1_SINK_MODE,
+	VOUT2,
+	IOUT2_SOURCE_MODE,
+	IOUT2_SINK_MODE,
+	VOUT3,
+	IOUT3_SOURCE_MODE,
+	IOUT3_SINK_MODE,
+	VOUT4,
+	IOUT4_SOURCE_MODE,
+	IOUT4_SINK_MODE,
+	VOUT5,
+	IOUT5_SOURCE_MODE,
+	IOUT5_SINK_MODE,
+	VOUT6,
+	IOUT6_SOURCE_MODE,
+	IOUT6_SINK_MODE,
+	VOUT7,
+	IOUT7_SOURCE_MODE,
+	IOUT7_SINK_MODE,
+	DIE_TEMPERATURE,
+	TIED_TO_AGND_INTERNALLY
+};
+
+/* By default all values are set to 0 */
+struct ad3530r_transfer_config {
+	/* Defines the length of the loop when streaming data */
+	uint8_t		stream_mode_length;
+	/* Determines Sequential Addressing Behavior */
+	uint8_t		addr_asc : 1;
+	/* Select Streaming or Single Instruction Mode */
+	uint8_t		single_instr : 1;
+	/* Determines the length of the address in the instruction phase */
+	uint8_t		short_instr : 1;
+	/*
+	 * Set this bit to prevent the STREAM_MODE LENGTH value from
+	 * automatically resetting to zero
+	 */
+	uint8_t		stream_length_keep_value : 1;
```

Comment:
We have only used bitfields in order to directly map a device register value to a set of "status flag" bits in a C struct. I would say they're not worth bothering with.